src/Domain/ElementId.cpp

```diff
@@ -18,24 +18,33 @@
 template <size_t VolumeDim>
 constexpr size_t ElementId<VolumeDim>::volume_dim;
 
+static_assert(std::is_pod<ElementId<1>>::value, "ElementId is not POD");
+static_assert(std::is_pod<ElementId<2>>::value, "ElementId is not POD");
+static_assert(std::is_pod<ElementId<3>>::value, "ElementId is not POD");
+
+static_assert(sizeof(ElementId<1>) == 1 * sizeof(int),
+              "Wrong size for ElementId<1>");
+static_assert(sizeof(ElementId<2>) == 2 * sizeof(int),
+              "Wrong size for ElementId<2>");
+static_assert(sizeof(ElementId<3>) == 3 * sizeof(int),
+              "Wrong size for ElementId<3>");
```

Comment:
1. Yes, it is to ensure that the class, even if the tests aren't compiled, meets the size requirements of Charm++. I prefer having them here since I found it useful when making changes to know that the exact constraints imposed by Charm++ are satisified.
2. I'm not sure exactly what you mean with the instantiations. If you are referring to the `constexpr size_t ElementId<VolumeDim>::volume_dim;`, that actually defines the member function so that when the class is instantiated, so is the member functions. If you mean something else I'll need you to clarify in order to be able to answer :)

Reply:
1. Fair enough, I am convinced — but would like a comment, as per my review

2. I mean: to apply the static_assert to, e.g., `ElementId<2>`, doesn't the compiler need to have already "stamped out" the code for the specific class `ElementId<2>` from the general template `ElementId<VolumeDim>`? And isn't this "stamping out" happening in the GENERATE_INSTANTIATIONS? I'm trying to better understand what order the compiler works through these lines of code, as it compiles the source file.

Reply:
1. :+1: sounds reasonable!
2. Ah, I see. So since it is a template, the compiler can do (and I guess must) an implicit instantiation. We could move the `static_assert`s into the instantiation calls. I believe the compiler goes top to bottom, so would create the implicit instantiations of what it needs, then the implicit instantiations. The only reason I have the `static_assert`s outside of the instantiation code is because I found it easier to read and understand, but I'm happy to move it if you think that's clearer. Does that answer your question(s)?

Reply:
2. Ah, the implicit instantiation makes sense. I definitely agree that the `static_asserts` are clearer (both to find and to understand the intent of) here than they would be in the instantiations, so I agree with the current placement. This does answer my question, thanks! 🙂